content/cookbooks/deployment/docker.md

```diff
@@ -0,0 +1,107 @@
+---
+datetime: 2022-02-02
+author: Roth Fessler Maxime
+avatarUrl: https://avatars.githubusercontent.com/u/57860498?v=4
+summary: Cookbook to deploy AdonisJS application to Docker
+---
+
+This guide covers the steps for deploying an AdonisJS application to a [Docker Image](https://www.docker.com).
+
+:::note
+This guide assumes you have a reasonably basic understanding of how Docker images and containers are built and deployed.
+:::
+
+Deploying AdonisJS to a docker image isn't much different from deploying a simple NodeJS application to a docker image, however it has a few extra steps due to its build step.
+
+You can build your project for production by running the following ace command. Learn more about the [TypeScript build process](../../guides/fundamentals/typescript-build-process.md)
+
+```sh
+node ace build --production
+# OR use the npm script
+npm run build
+```
+
+## Create the docker image
+
+:::note
+In the interest of simplicity this guide will only cover how to create a `Dockerfile` and not a `docker-compose.yaml` file.
+:::
+
+Before beginning make you sure you have docker engine version `17.05` or newer.
+
+```sh
+docker version
+```
+
+- In the root folder of your project create a `Dockerfile`
+- In the root folder also create a `.dockerignore` file
+- Add `node_modules` and `.env` file to the `.dockerignore` file as well as any other folders/files you don't want as part of the deployment
+
+```dockerignore
+// title: .dockerignore
+node_modules
+.env
+.git
+build
+```
+- Open your empty `Dockerfile` and paste the following inside of it
+
+```dockerfile
+# Pull latest NodeJs LTS version with Alpine Linux
+# and rename it "base" to re-use it for the next stages
+FROM node:16-alpine3.14 AS base
+# Make a working directory for your app in the container
+WORKDIR /home/app
+
+# Build stage for build
+FROM base AS builder
+# Add git, some node packages require git to be available
+RUN apk add --no-cache git
```

Comment:
Luckily `git` is no longer required :yum: 

It used to be requirement for `http-server` if I remember correctly, but now it seems to install without that too